selfservice/strategy/oidc/strategy_login.go

```diff
@@ -84,6 +82,11 @@ type UpdateLoginFlowWithOidcMethod struct {
 	//
 	// required: false
 	UpstreamParameters json.RawMessage `json:"upstream_parameters"`
+
+	// Only used in API-type flows, when an id token has been received by mobile app directly from oidc provider.
+	//
+	// required: false
+	IDToken string `json:"id_token"`
```

Comment:
What are the security implications when we get the ID token from the user instead of the OIDC provider?

Reply:
If I recall correctly, that should be fine. The ID token will be validated on our end and we will check both issuer and, more importantly, audience. By checking audience, we ensure that the ID token was issued on our behalf, and not someone elses behalf. That way we avoid the vulnerability that could be caused by instead accepting access tokens as proof of authentication, as access tokens don't have an audience and they could have been issued for any app.

Still, this allows a eve to trick adam into using eve's ID token. Would be good to know how others deal with that?